index/scorch/scorch.go

```diff
@@ -117,9 +121,30 @@ func NewScorch(storeName string,
 	return rv, nil
 }
 
+func (s *Scorch) waterMark() uint64 {
+	s.memLock.Lock()
+	w := s.highWaterMark
+	s.memLock.Unlock()
+	return w
+}
+
+func (s *Scorch) incrWaterMark() {
+	s.memLock.Lock()
+	s.highWaterMark++
+	s.memLock.Unlock()
+}
+
+func (s *Scorch) decrWaterMark() {
+	s.memLock.Lock()
+	s.highWaterMark--
```

Comment:
quick thing, on naming, a "high water mark" i think usually means some level or number that doesn't go down